includes/class-comments.php

```diff
@@ -0,0 +1,123 @@
+<?php
+namespace Activitypub;
+
+/**
+ * Comments Class
+ *
+ * @author Django
+ */
+class Comments {
+	/**
+	 * Initialize the class, registering WordPress hooks.
+	 */
+	public static function init() {
+
+		\add_filter( 'preprocess_comment', array( '\Activitypub\Comments', 'preprocess_comment' ) );
+		\add_filter( 'comment_post', array( '\Activitypub\Comments', 'postprocess_comment' ), 10, 3 );
+		\add_filter( 'wp_update_comment_data', array( '\Activitypub\Comments', 'comment_updated_published' ), 20, 3 );
+		\add_action( 'transition_comment_status', array( '\Activitypub\Comments', 'schedule_comment_activity' ), 20, 3 );
+		\add_action( 'edit_comment', array( '\Activitypub\Comments', 'edit_comment' ), 20, 2 );//schedule_admin_comment_activity
+		\add_filter( 'get_comment_text', array( '\Activitypub\Comments', 'comment_append_edit_datetime' ), 10, 3 );
+
+	}
+
+	/**
+	 * preprocess_comment()
+	 * preprocess local comments for federated replies
+	 */
+	public static function preprocess_comment( $commentdata ) {
+		// only process replies from local actors
+		if ( ! empty( $commentdata['user_id'] ) ) {
+			$commentdata['comment_type'] = 'activitypub';
```

Comment:
shouldn't this be `comment_type` be `comment` and the `protocol` comment-meta be `activitypub`? Otherwise we can not differ between boosts, shares and comments.

Reply:
I wrote that part of the code years ago 😅

It doesn't seem to interfere with likes and shares in `handle_reaction`, 
```
'comment_type' => \esc_attr( \strtolower( $object['type'] ) ),
```

I think the original concern was how to only process/federate replies from authors. I will review and refactor this. This also gives me an idea for a Test Case

Reply:
@pfefferle For defining whose reply can federate, do you have a preference between `$user->roles`(ex. Author) & `$user->has_cap`(ex. `publish_posts`)? Capabilities are more flexible where there are custom roles, but could also have unintended outcomes.